blocks/library/image/index.js

```diff
@@ -106,6 +111,39 @@ registerBlockType( 'core/image', {
 					icon="format-image"
 					label={ __( 'Image' ) }
 					className={ className }>
+					<DropZone
```

Comment:
>Would it make sense to make an "ImageDropZone" component to abstract this away?

Maybe. Prior to 52c848c0d19ca5a49f88430bd80aa117cf714466 there'd been a `MediaDropZone` component for inserting new image blocks. In this case, the behavior sets attributes from within the image block. To make this generic we might need to expose a way to map properties of a media entity to attributes of the block.

In general I think the image block could do for some separating into smaller components, but within its own folder/scope, not general purpose components.